catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/netty/UdpStreamProcessor.java

```diff
@@ -325,22 +335,53 @@ public void shutdown() {
         }
     }
 
+    public ExecutorService getRolloverThreadPool() {
+        return rolloverThreadPool;
+    }
+
+    public void setRolloverThreadPool(ExecutorService executorService) {
+        this.rolloverThreadPool = executorService;
+    }
+
     public void checkForRollover() {
         packetBuffer.rotate(rolloverCondition)
                 .ifPresent(this::doRollover);
     }
 
-    public void doRollover(File tempFile) {
-        LOGGER.debug("performing video chunk rollover: tempFile={}", tempFile);
-        try {
-            rolloverAction.doAction(tempFile);
-        } catch (RolloverActionException e) {
-            LOGGER.debug("unable handle rollover file: tempFile={}", tempFile, e);
-        } finally {
-            if (!tempFile.delete()) {
-                LOGGER.debug("unable to delete temp file: filename={}", tempFile);
+    public Future<Void> doRollover(File tempFile) {
+
+        rolloverTaskCount.incrementAndGet();
+        LOGGER.debug(
+                "submitting rollover task to thread pool: submitted task count={} running task count={}",
+                rolloverTaskCount.get(),
```

Comment:
The value of `rolloverTaskCount` could already have changed since incremented on line 353 and the value of `rolloverThreadCount` would likely be different as well (especially as you've not submitted the task to the pool yet). Granted, this is only a snapshot debug message, but it might be good to indicate that the values being logged may differ from the moment-to-moment reality.